snapcraft/internal/project_loader/_config.py

```diff
@@ -108,6 +108,13 @@ def __init__(self, project_options=None):
         _ensure_grade_default(snapcraft_yaml, self._validator.schema)
 
         self.data = self._expand_env(snapcraft_yaml)
+        self._project_options.info = snapcraft.ProjectInfo(
+            name=self.data['name'],
+            version=self.data['version'],
+            summary=self.data.get('summary', ''),
+            description=self.data.get('description', ''),
+            confinement=self.data['confinement'],
```

Comment:
Why leave extracting the relevant information to the caller? `ProjectInfo` could just be handed the YAML and extract stuff on its own.

Reply:
I thought that's the style we prefer, but no strong opinion. I'm fine with handling extraction in the constructor.